python/torch_mlir_e2e_test/test_suite/scalar.py

```diff
@@ -151,3 +151,123 @@ def forward(self, lhs, rhs):
 @register_test_case(module_factory=lambda: CeilFloatModule())
 def CeilFloatModule_basic(module, tu: TestUtils):
     module.forward(torch.rand(()).double(), torch.rand(()).double())
+
+
+# ==============================================================================
+
+
+class BoolFloatFalseModule(torch.nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([], torch.float64, True),
+    ])
+    def forward(self, a):
+        sub = float(a) - float(a)
+        return bool(torch.ops.aten.Bool(float(sub)))
+
+
+@register_test_case(module_factory=lambda: BoolFloatFalseModule())
+def BoolFloatFalseModule_basic(module, tu: TestUtils):
+    module.forward(torch.rand(()).double())
+
+
+class BoolFloatTrueModule(torch.nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([], torch.float64, True),
+    ])
+    def forward(self, a):
+        return bool(torch.ops.aten.Bool(float(a)))
+
+
+@register_test_case(module_factory=lambda: BoolFloatTrueModule())
+def BoolFloatTrueModule_basic(module, tu: TestUtils):
+    module.forward(torch.rand(()).double())
+
+
+class BoolFloatConstantModule(torch.nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+    ])
+    def forward(self):
+        return bool(torch.ops.aten.Bool(5.0))
+
+
+@register_test_case(module_factory=lambda: BoolFloatConstantModule())
+def BoolFloatConstantModule_basic(module, tu: TestUtils):
+    module.forward()
+
+
+# ==============================================================================
+
+
+class BoolIntFalseModule(torch.nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([], torch.int64, True),
+    ])
+    def forward(self, a):
+        sub = int(a) - int(a)
+        return bool(torch.ops.aten.Bool(int(sub)))
+
+
+@register_test_case(module_factory=lambda: BoolIntFalseModule())
+def BoolIntFalseModule_basic(module, tu: TestUtils):
+    module.forward(torch.randint(100, ()))
+
+
+class BoolIntTrueModule(torch.nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([], torch.int64, True),
+    ])
+    def forward(self, a):
+        return bool(torch.ops.aten.Bool(int(a)))
+
+
+@register_test_case(module_factory=lambda: BoolIntTrueModule())
+def BoolIntTrueModule_basic(module, tu: TestUtils):
+    module.forward(torch.randint(100, ()))
```

Comment:
`torch.randint` by default has `0` as its lower bound. Can you specify a higher lower bound so that it doesn't accidentally output `False`? (For both the `BoolFloatTrue` and `BoolIntTrue`, the test is not gonna fail if the input is zero, but just to be consistent with the name of the test)